src/Block.cpp

```diff
@@ -298,8 +338,14 @@ Block::Output(std::ostream &out, FactMgr* fm, int indent) const
 	if (CGOptions::math_notmp())
 		OutputTmpVariableList(out, indent);
 
+
 	OutputVariableList(local_vars, out, indent);
-	OutputStatementList(stms, out, fm, indent);
+
+	if(CGOptions::check_local()){
+		OutputStatementListCheckLocal(stms, local_vars, out, fm, indent);
```

Comment:
Is option ```check_local``` only useful for float-test? If so, we can also add ```CGOptions::float_test()``` to the ```if``` condition to ensure we wouldn't accidentally invoke ```OutputStatementListCheckLocal```?